### ISSUE TITLE
Several (optional?) dependencies found...

### ISSUE THREAD

**Issue by dothebart** [Valloric/ycmd#928]
On debian buster: 

    apt-get install python-flake8 libmono-microsoft-csharp4.0-cil libmono-system-data-datasetextensions4.0-cil libicsharpcode-nrefactory-cecil5.0-cil libicsharpcode-nrefactory-cil-dev libicsharpcode-nrefactory-csharp5.0-cil libicsharpcode-nrefactory-ikvm5.0-cil libicsharpcode-nrefactory-xml5.0-cil libicsharpcode-nrefactory5.0-cil

but still wasn't able to get c# completion working - whats actually required here?

**Comment by micbou** [comment]
You need to [install mono](http://www.mono-project.com/download/stable/#download-lin-debian) then run the `build.py` script with the `--cs-completer` option:
```sh
./build.py --cs-completer
```

**Comment by dothebart** [comment]
So you list in the page: 

    sudo apt-get install mono-xbuild for C#.

which should rather be 

    sudo apt-get install mono-xbuild  mono-devel

;-)

**Comment by micbou** [comment]
You are right. Just tried it and I had to install `mono-devel`. Since `mono-xbuild` is a dependency of `mono-devel`, `sudo apt install mono-devel` should be enough. PR #929 updates the README accordingly. Thanks!

**Comment by dothebart** [comment]
python-flake8 is only needed for testing? I don't think its mentioned on the README either...

**Comment by micbou** [comment]
Yes, only for testing and it's supposed to be installed through the `test_requirements.txt` file:
```sh
pip install -r test_requirements.txt
```

**Comment by dothebart** [comment]
ok, tnx.
